src/compile/facet.ts

```diff
@@ -219,14 +219,22 @@ export class FacetModel extends ModelWithField {
   protected assembleDefaultLayout(): VgLayout {
     const columns = this.channelHasField('column') ? this.columnDistinctSignal() : 1;
 
-    // TODO: determine default align based on shared / independent scales
+    let align: LayoutAlign = 'all';
+
+    // Do not align the cells if the scale corresponding to teh directin is indepent.
+    // We always align when we facet into row and column.
```

Comment:
The comment "We always align when we facet into row and column." doesn't match what you do below.

Reply:
Better? "We always align when we facet into both row and column."